### PR TITLE
feat(cdf-typescript): Revise package manifest

### PR DESCRIPTION
**Pull request by cwang**
## Description

- Revise package manifest to include cloud vendor support info
- Add schema gen step in releasing workflow

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor and code improvement (non-breaking change which improves code quality and/or performance)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] Tests
- [x] Other chores such as maintenance

## How Has This Been Tested?

Sample json files used for testing package manifest have been updated to cater for new format.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] The commit message follows the convention of this project

### CODE REVIEW

File: core/typescript/package.json [hunk 1]
```diff
@@ -51,7 +51,8 @@
     "doc:json": "typedoc src/ --exclude **/*.spec.ts --json build/docs/typedoc.json",
     "sitedoc": "typedoc --plugin typedoc-plugin-markdown src/ --exclude **/*.spec.ts --out ../../docs/references/typescript/${npm_package_version}",
     "reset-hard": "git clean -dfx && git reset --hard && npm i",
```
Comment (janaka) [inline core/typescript/package.json:53]: curious: assume release please or the like needed to do this?

File: core/typescript/src/lib/domain.ts [hunk 2]
```diff
@@ -13,10 +13,29 @@
  * @group For both project creators and package authors
```
Comment (janaka) [inline core/typescript/src/lib/domain.ts:13]: "Project Creators" aka App Developers, there's no third user persona right? should replace. Project Creator is very  FABR context specific to understand compared to App Developer which is intuitive.

**Comment by janaka** [comment]
are `metadata-minimal.spec.json` and `metadata-normal.spec.json` templates or something?

**Review by janaka — APPROVED** [review]
are `metadata-minimal.spec.json` and `metadata-normal.spec.json` templates or something?

File: core/typescript/src/lib/domain.ts (hunk 2 above)
Comment (cwang) [reply core/typescript/src/lib/domain.ts:13]: I'm neutral on this one. On one hand, exactly as you said in order to standardise the terms we use, on the other I'm relaxed about using different sets of terms between OSS and SaaS where we talk more sharply about specific persons that require accurate messaging. 
I think we should merge this in, and do a global search-n-replace if needed later.

**Comment by cwang** [comment]
> are `metadata-minimal.spec.json` and `metadata-normal.spec.json` templates or something?

They're for testing, the `.spec.` gives it away. The way this project is set up is that test and source are mixed together in one place.

**Pull request merged**